eisbuk-admin/src/components/layout/AttendanceSheet.tsx

```diff
@@ -0,0 +1,74 @@
+import * as React from "react";
+import Table from "@material-ui/core/Table";
+import TableBody from "@material-ui/core/TableBody";
+import TableCell from "@material-ui/core/TableCell";
+import TableContainer from "@material-ui/core/TableContainer";
+import TableHead from "@material-ui/core/TableHead";
+import TableRow from "@material-ui/core/TableRow";
+import Paper from "@material-ui/core/Paper";
+import { fb2Luxon } from "@/utils/date";
+import { CustomerWithAttendance, SlotInterface } from "@/types/temp";
+
+export interface Props extends SlotInterface {
+  /**
+   * Record of customers who have booked (or are manually as attended),
+   * keyed by `customerId` and having all of the data for the customer plus
+   * values for `bookedInterval` and `attendedInterval`.
+   */
+  customers: CustomerWithAttendance[];
+}
+// const groupCustomersByAttendedInterval = (
+//   intervals: {
+//     [key: string]: SlotInterval;
+//   },
+//   customers: CustomerWithAttendance[]
+// ) => {
+//   const intervalsEmpty = Object.keys(intervals).reduce((acc, curr) => {
+//     acc[curr] = [];
+//     return acc;
+//   }, {});
+//   customers.forEach((customer) => {
+//     customer.bookedInterval &&
+//       intervalsEmpty[customer.bookedInterval].push(customer.name);
+//   });
+//   return intervalsEmpty;
+// };
+
+const AttendanceSheet: React.FC<Props> = ({ customers, intervals, date }) => {
+  // const customersGroupedByInterval = groupCustomersByAttendedInterval(
+  //   intervals,
+  //   customers
+  // );
+
+  return (
+    <TableContainer component={Paper}>
+      <Table aria-label="simple table">
+        <TableHead>
+          <TableRow>
+            <TableCell>
+              Attendance for {fb2Luxon(date).toFormat("EEEE dd-MM-y")}
+            </TableCell>
+          </TableRow>
+        </TableHead>
+        {Object.keys(intervals).map((interval) => (
+          <TableBody>
+            <TableRow key={interval}>
```

Comment:
If you wanted to render `TableBody` for each interval (and you don't), you would key the `<TableBody>` component, not the `<TableRow>`.

_This is because, in the component tree, `TableBody` is a node within the component list (rendered by map function), with the children being child nodes of `TableBody` node,  and React Fiber only cares for keys of the list nodes (`TableBody` in this case)._

However, you don't want to render `<TableBody>` for each entry, but rather have:
```jsx
<TableBody>
  {intervals.map(() => 
    <TableRow key={interval}>
      ...
    </TableRow>
  }
</TableBody>
```   
**Things to keep in mind:**
- you only key the top level nodes in mapped list of components _(there is advanced use for keys, but no need to get into it now)_
- always check what you want to render as a list and what you want to render only once (as a list container)

Reply:
Right...now I see you're rendering multiple child nodes for each map function result. In those cases, it makes sense to use react fragment `<></>` to wrap all children: this is due to the rule that the mapping function should return only one component (only one fiber node) and in this case the fragment becomes that one node, and the components you're returning are the child nodes of that (fragment) node. 
Furthermore, since you want to key each node of the mapping function, here you should use `<React.Fragment key={interval} >` as the `React.Fragment` is the direct child of mapped list of components (`React.Fragment` is the same as `<>` but it accepts `key` prop unlike the short `<>` notation.